src/main/java/com/nutomic/syncthingandroid/util/Logger.java

```diff
@@ -0,0 +1,54 @@
+package com.nutomic.syncthingandroid.util;
+
+import android.widget.TextView;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+public class Logger {
+
+    private final static String TAG = "Logger";
+
+    private LinkedList<String> mLog;
+    private int mLimit;
+
+    public Logger(int limit) {
+        mLog = new LinkedList<>();
+        mLimit = limit;
+    }
+
+    public synchronized void log(String str) {
+        while (mLog.size() > mLimit) { mLog.remove(); }
+        mLog.add(str);
+    }
+
+    public synchronized String getLog() {
+        StringBuilder sb = new StringBuilder();
+        Iterator it = mLog.iterator();
+        while (it.hasNext()) {
+            sb.append(it.next());
+            sb.append("\n");
+        }
+        return sb.toString();
+    }
+
+    public String getAndroidLog() {
+        try {
+            Process process = Runtime.getRuntime().exec("logcat -d '*:d'");
```

Comment:
I'd do `SyncthingNativeCode:s`, or we have that twice. Though it may also be useful, not sure.

Also, comment your class/methods :p